### ISSUE TITLE
listen +port prevents bot connections

### ISSUE THREAD

**Issue by vanosg** [eggheads/eggdrop#754]
Placeholder to troubleshoot this issue from discussion in #552  .

A bot listening with +port isn't sending a STARTTLS. 
Instead, the link fails and the initiating bot sends/receives:
```
[06:42:15] [m->testbot] goodbot
[06:42:15] [m<-testbot] ��
[06:42:15] [m<-testbot] 
[06:42:15] [m<-testbot] Nickname.
[06:42:15] Refused telnet@scoobalooba (invalid handle: goodbot)
```
and the connection fails.

**Comment by michaelortmann** [comment]
Are you sure, the listener is the problem or the right place to fix? In case #552 the caller was not doing STARTTLS, if it did, it would probably succeed. There is already theory of a patch in #552, and a small example demostrating that fix could work. That fix opened a new problem that must be fixed, a fallback from STARTTLS to PLAIN if the connection failed. There is also a theory of a patch in #552 for this addional problem.

You can find the fix and the rest of the story in #755.

**Comment by michaelortmann** [comment]
Its interesting that in some cases the calling bot does start the handshake and in other cases the called bot sends starttls. Maybe there are cases that both is happening. Something doesnt look right here. I realy wasnt expecting a called bot to send starttls if i didnt initialize it. This could lead to bigger problems like in #552 where we want to force into non ssl link. We dont want and dont need the called bot to send unwanted starttls!

Part of the problem here is also that we speak of 2 thinks and think in different ways, when we take into account the called bot could send starttls. i really didn't think of this possibility until i was forced to when i had to look deeper into #552.

**Comment by vanosg** [comment]
in tcldcc.c, when establishing a listen port:
```
#ifdef TLS
  if (*argv[1] == '+')
    dcc[idx].ssl = 1;
```

then, when receiving an incoming connection:
dcc.c:
```
    if (!dcc[idx].ssl) {
      dprintf(idx, "starttls\n");
```

by setting ssl to 1, starttls is not set.

**Comment by michaelortmann** [comment]
If we patch the line you just found, it would probably fix a problem for patched bots, but it wont fix it for unpatched 1.8.3 that have listen +PORT set. My patch also covers this problem, it doesnt need the starttls of the called bot to ssl link.
